### PR TITLE
Add ADAL01

### LINKED ISSUE #37: Add ADAL01
Template design was completed on GitLab but never added to TLG-C.

### PR DESCRIPTION
**Pull request by edelarua**
Work from https://github.com/insightsengineering/tlg-catalog/pull/38 moved here due to repo restructuring since previous PR was created.

Closes #37

### CODE REVIEW

**Comment by edelarua** [comment]
Hi @qiliu1013,

After your refactor of ADAB in `random.cdisc.data`, the "treatment emergent status" data is no longer present in `adab` (see change [here](https://github.com/insightsengineering/random.cdisc.data/commit/df423fcf7c2f47ca28518fa3ebee837e01d5115d#diff-976aa496cbf3a6a35f575918a4ee7b0621aa8a6c522ce487c868625e592c7db1R88)), and the `ATACHAR` variable is no longer present in `adab` (see change [here](https://github.com/insightsengineering/random.cdisc.data/commit/df423fcf7c2f47ca28518fa3ebee837e01d5115d#diff-976aa496cbf3a6a35f575918a4ee7b0621aa8a6c522ce487c868625e592c7db1L165)). Consequently, the "Characterization Results", "Patient Treatment Emergent ADA Status", and "Patient Treatment Emergent NAb Status" columns can no longer be derived in this listing. How should we proceed?

I have commented out the broken parts of the code to create the listing for now, which generates ADAL01 without the four missing columns until this is fixed.

I will be out of office for the upcoming week but if you have any questions please feel free to reach out to anyone on the SME team!

**Comment by qiliu1013** [comment]
> Hi @qiliu1013,
> 
> After your refactor of ADAB in `random.cdisc.data`, the "treatment emergent status" data is no longer present in `adab` (see change [here](https://github.com/insightsengineering/random.cdisc.data/commit/df423fcf7c2f47ca28518fa3ebee837e01d5115d#diff-976aa496cbf3a6a35f575918a4ee7b0621aa8a6c522ce487c868625e592c7db1R88)), and the `ATACHAR` variable is no longer present in `adab` (see change [here](https://github.com/insightsengineering/random.cdisc.data/commit/df423fcf7c2f47ca28518fa3ebee837e01d5115d#diff-976aa496cbf3a6a35f575918a4ee7b0621aa8a6c522ce487c868625e592c7db1L165)). Consequently, the "Characterization Results", "Patient Treatment Emergent ADA Status", and "Patient Treatment Emergent NAb Status" columns can no longer be derived in this listing. How should we proceed?
> 
> I have commented out the broken parts of the code to create the listing for now, which generates ADAL01 without the four missing columns until this is fixed.
> 
> I will be out of office for the upcoming week but if you have any questions please feel free to reach out to anyone on the SME team!

Hi @edelarua, no need to show "Characterization Results" and "Patient Treatment Emergent NAb Status" in the ADAL01. A the new standard specs, there were move into PARAM now. We don't need to consider them.
"Patient Treatment Emergent NAb Status" now was in PARAMCD = "ADASTAT1". 

Did I answer all your questions?

Thanks,
Qi

**Comment by edelarua** [comment]
> Did I answer all your questions?

To confirm - "Patient Treatment Emergent ADA Status" also no longer need to be considered, correct? If so then yes you answered all my questions, and the template should be ready for review! Thanks Qi!!

**Comment by qiliu1013** [comment]
> > Did I answer all your questions?
> 
> To confirm - "Patient Treatment Emergent ADA Status" also no longer need to be considered, correct? If so then yes you answered all my questions, and the template should be ready for review! Thanks Qi!!

Sorry. Patient Treatment Emergent ADA Status is PARAMCD = "ADASTAT1". No need to consider "Patient Treatment Emergent NAb Status"

**Comment by qiliu1013** [comment]
Hi @edelarua, can you please share the output with me? Thanks!

**Comment by shajoezhu** [comment]
hi @qiliu1013 , can you review the output at https://github.com/insightsengineering/tlg-catalog/pull/116/files#diff-800dcce08155543ad547403b286591745b8c15d6d35a7b4fec7dc33a51da067c please

**Comment by edelarua** [comment]
@qiliu1013 see a screenshot of the TLG-C output here:

adal01_screenshot

File: package/tests/testthat/_snaps/adal01/markdown-snaps.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# lsting
+
+```         
+Code
+  print(data_snap[[i]])
+Output
+  # A tibble: 24 × 9
+     USUBJID               VISIT      ISTPT   NFRLT RES_ADA  TU_ADA RES_NAB  TU_NAB AVAL_pk
+     <lstng_ky>            <lstng_ky> <fct>   <dbl> <chr>    <chr>  <chr>    <chr>  <chr>  
+   1 AB12345-RUS-16-id-165 Day 2      24H        24 Negative N/A    NA       N/A    0.00
```
Comment (qiliu1013) [inline package/tests/testthat/_snaps/adal01/markdown-snaps.md:10]: @edelarua, do you know why AVAL_PK is 0.00? It should be some numbers or NA.

File: book/listings/pharmacokinetic/adal01.qmd [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    USUBJID = "Subject ID",
+    VISIT = "Visit",
+    ISTPT = "Timepoint",
+    NFRLT = "Nominal\nTime\n(hr)",
+    RES_ADA = "Sample\nADA\nResult",
+    TU_ADA = "ADA\nTiter\nUnits\n(1)",
+    RES_NAB = "Sample\nNeutralizing\nAntibody\n(NAb) Result",
+    TU_NAB = "NAb\nTiter\nUnits\n(2)",
+    AVAL_pk = paste0("Drug\nConcentration\n(", conc_u, ") (3)")
+  ) %>%
```
Comment (qiliu1013) [inline book/listings/pharmacokinetic/adal01.qmd:90]: @edelarua, patient treatment emergent ADA status PARAMCD = 'ADASTAT1' and Patent Treatment Emergent NAB status PARAMCD = 'ADASTAT2' are missing. I may confused you last time. We had all the columns below except the column "Charcterization Results".
Screen Shot 2023-09-20 at 2 50 42 PM

File: package/tests/testthat/_snaps/adal01/markdown-snaps.md (hunk 1 above)
Comment (edelarua) [reply package/tests/testthat/_snaps/adal01/markdown-snaps.md:10]: @qiliu1013 I believe those `AVAL` values were set to 0 in `adpc` in `random.cdisc.data`. I have added a line in to convert 0 values as well as `NA`s to `"N/A"` for this column.

File: book/listings/pharmacokinetic/adal01.qmd (hunk 2 above)
Comment (edelarua) [reply book/listings/pharmacokinetic/adal01.qmd]: @qiliu1013 in that case [my earlier comment](https://github.com/insightsengineering/tlg-catalog/pull/116#issuecomment-1712110339) still applies - values for these two columns are not generated within `random.cdisc.data` so these datasets will need to be updated if we want to include these two columns in the table.

**Comment by shajoezhu** [comment]
blocking this for the moment. It seems we need some refactoring in the rcd data again.

**Comment by qiliu1013** [comment]
@shajoezhu and @edelarua, can you please let me know where to find the latest adal01 code? I didn't see it under 37_adal01 branch. Thanks!

**Comment by shajoezhu** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by qiliu1013** [comment]
I have read the CLA Document and I hereby sign the CLA

File: book/listings/pharmacokinetic/adal01.qmd [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+```{r setup, message=FALSE}
+#| code-fold: show
+
+library(dplyr)
+library(rlistings)
+library(scda)
+
+adpc <- synthetic_cdisc_dataset("latest", "adpc")
+adab <- synthetic_cdisc_dataset("latest", "adab")
```
Comment (shajoezhu) [inline book/listings/pharmacokinetic/adal01.qmd:21]:
```suggestion
library(random.cdisc.data)

adpc <- cadpc
adab <- cadab
```

**Review by shajoezhu — CHANGES_REQUESTED** [review]
need to update snapshot

**Comment by shajoezhu** [comment]
hi @edelarua , could you update the snapshot, once https://github.com/insightsengineering/random.cdisc.data/pull/317 is in please. Thanks!

**Comment by shajoezhu** [comment]
this is potentially blocked by https://github.com/insightsengineering/random.cdisc.data/issues/319

**Comment by edelarua** [comment]
Snapshot has been updated.

**Review by shajoezhu — APPROVED** [review]
lgtm! thanks @edelarua

File: book/listings/pharmacokinetic/adal01.qmd [hunk 4]
```diff
@@ -0,0 +1,152 @@
+---
+title: ADAL01
+subtitle: Listing of Anti-Drug Antibody Data for Patients with At Least One ADA Sample Datum by Treatment
+---
+
+------------------------------------------------------------------------
+
+{{< include ../../test-utils/envir_hook.qmd >}}
```
Comment (shajoezhu) [inline book/listings/pharmacokinetic/adal01.qmd:8]:
```suggestion
{{< include ../../_utils/envir_hook.qmd >}}
```

File: book/listings/pharmacokinetic/adal01.qmd [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+:::
+
+## Data Setup
+
+```{r setup}
+#| code-fold: show
+```
+::::
+
+{{< include ../../test-utils/save_results.qmd >}}
```
Comment (shajoezhu) [inline book/listings/pharmacokinetic/adal01.qmd:150]:
```suggestion
{{< include ../../_utils/save_results.qmd >}}
```

**Comment by shajoezhu** [comment]
blocked by #270

**Pull request merged**